### DISCUSSION TITLE
nncase 2.1 RuntimeError: Failed to get hostfxr path.

### DISCUSSION THREAD

**Discussion by Blatts01** [kendryte/nncase#1013] [Q&A] [answered]
Hello, 

I tried to install the prebuilt binary (Linux and OSx) but I got the following error in both cases: 

.venv/lib/python3.9/site-packages/nncase/__init__.py", line 43, in _initialize
    _nncase.initialize(compiler_path)
RuntimeError: Failed to get hostfxr path.

The installer runs without any errors.

To reproduce (OSX): 

ProductName:	macOS
ProductVersion:	12.6.7
BuildVersion:	21G651

(.venv) (base) $ python
Python 3.9.6 (default, Oct 18 2022, 12:41:40) 
[Clang 14.0.0 (clang-1400.0.29.202)] on darwin

pip install nncase-2.1.0.20230703-cp39-cp39-macosx_10_15_x86_64.whl 

Sadly, the docker image was not available, the target hardware is a k210 board. 

Best Regards

**Comment by curioyang** [comment, marked as answer]
Q1 :`RuntimeError: Failed to get hostfxr path.`
You need to install dotnet-7.0.
In ubuntu, use `sudo apt-get install dotnet-sdk-7.0`. 
If you still have problems after installation, maybe you install dotnet in a virtual enviroment, set the environment variables `export DOTNET_ROOT=/usr/share/dotnet`.

Q2: `Sadly, the docker image was not available, the target hardware is a k210 board.`
Unfortunately, nncase-v2 does not support k210. You need v0.2 to get good performance. It supports more operations in v1, but performance is poor.
